src/negativity.jl

```diff
@@ -89,7 +89,7 @@ function _partial_transpose(ρ::QuantumObject{<:AbstractArray, OperatorQuantumOb
         [pt_dims[n, 3 - mask2[n]] for n in 1:nsys]  # opposite value in mask2 (1 -> 2, and 2 -> 1)
     ]
     return QuantumObject(
-        reshape(PermutedDimsArray(reshape(ρ.data, (ρ.dims..., ρ.dims...)), pt_idx), size(ρ)),
+        Matrix(reshape(PermutedDimsArray(reshape(ρ.data, (ρ.dims..., ρ.dims...)), pt_idx), size(ρ))),
```

Comment:
I don’t remember if this function is gpu compatible. If yes, should ve convert it into the corresponding type?

Reply:
This is pretty wierd.
if I use: `reshape(PermutedDimsArray(reshape(ρ.data, (ρ.dims..., ρ.dims...)), pt_idx), size(ρ))`, it returns something in the type of:
```
Base.ReshapedArray{Float64, 2, PermutedDimsArray{Float64, 4, (1, 4, 3, 2), (1, 4, 3, 2), Array{Float64, 4}}, Tuple{Base.MultiplicativeInverses.SignedMultiplicativeInverse{Int64}, Base.MultiplicativeInverses.SignedMultiplicativeInverse{Int64}, Base.MultiplicativeInverses.SignedMultiplicativeInverse{Int64}}}
```

But if I use: `reshape(permutedims(reshape(ρ.data, (ρ.dims..., ρ.dims...)), pt_idx), size(ρ))`, it just returns in the type of `Matrix{Float64}`.

This causes type-instability, maybe we should just use `permutedims` (as I updated in ) in this case, or do you have other ideas that would work?

Reply:
Ok